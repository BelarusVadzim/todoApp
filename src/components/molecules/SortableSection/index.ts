import SortableSection from './SortableSection';

export { SortableSection };
