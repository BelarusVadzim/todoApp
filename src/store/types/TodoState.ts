import { TodoNote } from 'types';

export type TodoState = {
  todos: TodoNote[];
  filter?: string;
  pending?: boolean;
};
