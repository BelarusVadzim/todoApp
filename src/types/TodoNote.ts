export type TodoNote = {
  id?: number;
  done?: boolean;
  text: string;
};
