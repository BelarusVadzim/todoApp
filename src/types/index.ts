export * from './PropsWithClassName';
export * from './TodoNote';
