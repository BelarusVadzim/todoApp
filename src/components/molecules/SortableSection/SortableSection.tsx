import React from 'react';
import { PropsWithClassName } from 'types';
import { SortableContainer } from 'react-sortable-hoc';
import { useTodoStateService } from 'hooks';
import { SortableTodoItem } from '../SortableTodoItem';

const SortableSection: React.FC<PropsWithClassName> = () => {
  const { todos } = useTodoStateService();

  return (
    <div>
      {todos.map((value, index) => (
        <SortableTodoItem key={`item-${index}`} index={index} note={value} />
      ))}
    </div>
  );
};

export default SortableContainer<PropsWithClassName>(SortableSection);
