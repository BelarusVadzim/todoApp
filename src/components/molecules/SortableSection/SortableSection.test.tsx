import { render } from '@testing-library/react';
import { TodoNote } from 'types';
import SortableSection from './SortableSection';

const todo1 = { text: 'todo1' };
const todo2 = { text: 'todo2' };
const todos: TodoNote[] = [todo1, todo2];

jest.mock('hooks', () => ({
  useTodoStateService: () => ({
    todos,
  }),
}));

jest.mock('../SortableTodoItem', () => ({
  SortableTodoItem: jest.fn(({ note }: { key: string; note: TodoNote }) => {
    return <div>{note.text}</div>;
  }),
}));

describe('<SortableSection />', () => {
  it('should render properly', () => {
    const component = render(<SortableSection />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
