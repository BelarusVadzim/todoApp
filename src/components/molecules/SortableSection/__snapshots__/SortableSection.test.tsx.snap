// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortableSection /> should render properly 1`] = `
<DocumentFragment>
  <div>
    <div>
      todo1
    </div>
    <div>
      todo2
    </div>
  </div>
</DocumentFragment>
`;
